### DISCUSSION TITLE
Training KB descriptions

### DISCUSSION THREAD

**Discussion by Joozty** [explosion/spaCy#7205] [Help: Best practices]
I noticed @svlandeg trained descriptions [in her Wikipedia EL script](https://github.com/explosion/projects/blob/master/nel-wikipedia/train_descriptions.py) before inserting them into KnowledgeBase. What's the reason behind it? Does it bring more accurate results?

**Comment by svlandeg** [comment]
Training the descriptions is actually a crucial part of the current implementation of the entity linker. The descriptions are taken from Wikidata, and contain a lot of dense information. Basically, the EL algorithm compares the encodings of these descriptions to the sentences in which the mention occurs, to determine whether the context is similar.

You can find more information in [this video given at spaCy IRL 2019](https://www.youtube.com/watch?v=PW3RJM8tDGo&list=PLBmcuObd5An4UC6jvK_-eSl6jCvP1gwXc&index=6), or in the [more recent video tutorial](https://www.youtube.com/watch?v=8u57WSXVpmw).

**Reply by Joozty** [reply]
Thanks a lot. One last thing to clarify. If I understood you correctly it means that when I train my model I can remove all sentences that don't contain any entity from my corpus?

When the context in EL training is taken into account it just compares the similarity of the sentence where the entity is mentioned with vectors defined in KB, right?

**Reply by Joozty** [reply]
Oh, I see there is [`n_sents`](https://spacy.io/api/entitylinker#config) param to configure how many neighbours sentences take into account.

**Comment by Joozty** [comment]
I don't know where the accept button for the answer went. Maybe I don't have permission :thinking:. Nonetheless, I guess this can be marked as answered. Thanks

**Reply by svlandeg** [reply]
Some of the subforums have an "open-ended" style conversation, like this "best practices" subforum. These don't have a notion of "accepted answers". Nevertheless good to hear that it's resolved!

**Comment by Joozty** [comment]
Maybe I would have one more question for you @svlandeg if you don't mind. What's the rationale behind creating custom [word embeddings](https://github.com/explosion/projects/blob/master/nel-wikipedia/train_descriptions.py#L121) instead of using `doc.vector` property?

**Reply by svlandeg** [reply]
I don't remember all the history of the implementation details, but it would probably boil down to the same thing, yes!
